packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -643,3 +745,136 @@ class _SnackBarState extends State<SnackBar> {
     );
   }
 }
+
+// Hand coded defaults based on Material Design 2.
+class _SnackbarDefaultsM2 extends SnackBarThemeData {
+  _SnackbarDefaultsM2(BuildContext context)
+      : _theme = Theme.of(context),
+        _colors = Theme.of(context).colorScheme;
+
+  final ThemeData _theme;
+  final ColorScheme _colors;
```

Comment:
```suggestion
  final BuildContext context;
  late final ThemeData _theme = Theme.of(context);
  late final ColorScheme _colors = _theme.colorScheme;
  late final TextTheme _textTheme = theme.textTheme;
```

Making these late will allow them to only be computed in the case when they are needed.

Reply:
Nice - thanks